### ISSUE TITLE
Unable to discover the Google App Engine SDK root

### ISSUE THREAD

**Issue by NicolaSpreafico** [GoogleCloudPlatform/app-maven-plugin#104]
I have a pom.xml like follows:

```
<plugin>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>appengine-maven-plugin</artifactId>
    <version>1.0.0</version>
    <configuration>
        <deploy.version>${app.version}</deploy.version>
        <deploy.project>${app.id}</deploy.project>
        <deploy.promote>False</deploy.promote>
    </configuration>
</plugin>
```

I'm using Eclipse IDE with this version:

```
Eclipse Java EE IDE for Web Developers.
Version: Neon.1a Release (4.6.1)
Build id: 20161007-1200
```

While Eclipse is compiling the project, I'm getting this error in console:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
    at com.google.appengine.tools.info.ClassicSdk.getLoggingProperties(ClassicSdk.java:127)
    at com.google.appengine.tools.util.Logging.initializeLogging(Logging.java:35)
    at com.google.appengine.tools.enhancer.Enhance.main(Enhance.java:50)
Caused by: java.lang.RuntimeException: Unable to discover the Google App Engine SDK root. This code should be loaded from the SDK directory, but was instead loaded from file:/C:/Users/Nicola/.m2/repository/com/google/appengine/appengine-tools-sdk/1.9.44/appengine-tools-sdk-1.9.44.jar.  Specify -Dappengine.sdk.root to override the SDK location.
    at com.google.appengine.tools.info.SdkInfo.findSdkRoot(SdkInfo.java:117)
    at com.google.appengine.tools.info.SdkInfo.init(SdkInfo.java:238)
    at com.google.appengine.tools.info.SdkInfo.getSdkRoot(SdkInfo.java:192)
    at com.google.appengine.tools.info.SdkImplInfo.<clinit>(SdkImplInfo.java:20)
    ... 3 more
```

Is there a way to specify `sdk.root` as suggested?

**Comment by meltsufin** [comment]
If you take Eclipse out of the picture, and use `mvn clean install` in the terminal, does it successfully build the project?
@elharo

**Comment by NicolaSpreafico** [comment]
The problems seems to be related to `Datanucleus Enhancement`, so it is not the build itself that fails but the enhancement.

You can see from the title of the console Tab
![image](https://cloud.githubusercontent.com/assets/12278553/19602094/2e7585b4-97ac-11e6-9582-7cf1e8b8d90f.png)

If I disable the Enchancer the problems disappears
![image](https://cloud.githubusercontent.com/assets/12278553/19602135/49b8e2bc-97ac-11e6-9ebf-cf39140ab1d0.png)


I can keep the `Enchancer` disabled and carry on, but every time I do an "Update Maven Project" from Eclipse, the builder is automatically re-enabled (as another Builder, the previous one is still disabled)
![image](https://cloud.githubusercontent.com/assets/12278553/19602186/893890b8-97ac-11e6-8b8d-250e295b7152.png)

Because i do not use `DataNucleus` this is not a blocking problem, there is only the error log in the Eclipse console, but if there is a solution to configure the proper sdk-root or disable permanently the Enhancer from Maven build can be just fine.

**Comment by meltsufin** [comment]
It looks like you have an App Engine Plugin plugin for Eclipse. I'm not sure which one you're using, but the latest one is here: https://github.com/GoogleCloudPlatform/google-cloud-eclipse
Pinging people who work on it to see if they can help:
@patflynn @elharo @akerekes

**Comment by NicolaSpreafico** [comment]
Yes, I also have the Google Eclipse Plugin
`  Google Plugin for Eclipse 4.4/4.5/4.6	3.9.5.v20160930-1755	com.google.gdt.eclipse.suite.e44.feature.feature.group	Google, Inc.`

I am already aware of the latest Google Plugin but I have some opened issue so I haven't updated yet the plugin (so I'm using the previous WTP implementation offered by the plugin).

I'm waiting all the reported problem tol be fixed before switching the plugin, but if the problem is only related to this matter I can live with that (as I already pointed out, it is not a blocking problem due I'm not using datanucleus).

**Comment by meltsufin** [comment]
Just to clarify, if you remove the plugin from you pom.xml, the error goes away?

**Comment by NicolaSpreafico** [comment]
Yes, disabling the plugin there is no error at all in the Enhancement console
![image](https://cloud.githubusercontent.com/assets/12278553/19602980/abe7c946-97af-11e6-9242-286cc1cf24c0.png)

Also tried to enabling it back, and the error is shown again.

**Comment by meltsufin** [comment]
Interesting... So just adding this section to the pom, causes the error?
```
<plugin>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>appengine-maven-plugin</artifactId>
    <version>1.0.0</version>
    <configuration>
        <deploy.version>${app.version}</deploy.version>
        <deploy.project>${app.id}</deploy.project>
        <deploy.promote>False</deploy.promote>
    </configuration>
</plugin>
```

**Comment by NicolaSpreafico** [comment]
Yes, for the last test I simply commented out the plugin configuration and did a "Update Maven Project" to rebuild the project.

**Comment by meltsufin** [comment]
@NicolaSpreafico, we're unable to reproduce with a simple hello-world project. Can you share your complete `pom.xml` or a simple project that reproduces the issue? Thanks!

**Comment by akerekes** [comment]
@NicolaSpreafico Or if it's faster, can you share the `pom.xml`, the `.project` file and the contents of `<right click on project>/Properties/Maven/Lifecycle Mappings` when the new maven plugin is present in the pom file.

**Comment by NicolaSpreafico** [comment]
Attached you can find the test project I made to reproduce the issue
[test.zip](https://github.com/GoogleCloudPlatform/app-maven-plugin/files/547183/test.zip)

Here are the steps on how I created this project
1) Create a new Maven project with `appengine-standard-archetype` (I just used the `0.3.0` version)

2) Add the following plugin near the `com.google.cloud.tools:appengine-maven-plugin`
```
<plugin>
    <groupId>com.google.appengine</groupId>
    <artifactId>appengine-maven-plugin</artifactId>
    <version>${appengine.version}</version>
</plugin>
```
And... I think that is all. To trigger the error I did an `Update Maven Project` using Eclipse.

**Comment by elharo** [comment]
The problem here may be using

<groupId>com.google.appengine</groupId>

instead of

<groupId>com.google.cloud.tools</groupId>

Yes, the distinction is subtle and confusing. Notra culpa. Notra maxima culpa. We are working on fixing that, but as with all legacy products it's a long and drawn out process. :-(

**Comment by NicolaSpreafico** [comment]
I'm already well aware of it, this is not a confusion, I'm using both plugins with knowledge.
I already have a (still) open ticket with the Google Cloud Platform Support about this matter, but here is the explanation:

1) Using `appengine-standard-archetype` the project is created with `com.google.cloud.tools` plugin
For deploy and dev-server this plugin is enough.

2) For project management I also need to use `update_indexes`, `update_cron`, `update_queue`, ... which are available only for `com.google.appengine` plugin

So here is the explanation on why my pom.xml has both plugins configured. With the GCP Support I was doing some analysis on if using both plugins can generates conflicts, and it seems this is the first one I found.

**Comment by patflynn** [comment]
@NicolaSpreafico you can also use the com.google.cloud.tools plugin to update indexes, cron, etc by including them in your deployables.   Does that work for you?

**Comment by NicolaSpreafico** [comment]
@patflynn Is there a way to deploy ONLY indexes/cron/queue without deploying the entire application?
This is what I'm looking for.

I know that gcloud CLI provides this kind of behaviour, but I'm unable to reproduce it through Maven plugin

**Comment by patflynn** [comment]
@meltsufin @loosebazooka 

Can you guys put instructions on both the gradle and maven readme's on how to do this?
@lesv we should also add this to the cloud.google.com docs.

**Comment by NicolaSpreafico** [comment]
Any update on this? Have the documentation been integrated?

**Comment by akerekes** [comment]
Issue: #155
PR: #157

**Comment by akerekes** [comment]
Fixed in 1.3
